### PR TITLE
[compiler][be] Logger based debug printing in test runner

### PR DESCRIPTION
**Pull request by mofeiZ**
Avoid mutable logging enabled state and writing to `process.stdout` within our babel transform.

### CODE REVIEW

File: compiler/apps/playground/__tests__/e2e/__snapshots__/page.spec.ts/01-user-output.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
-function TestComponent(t0) {
+import { c as _c } from "react/compiler-runtime";
```
Comment (mofeiZ) [inline compiler/apps/playground/__tests__/e2e/__snapshots__/page.spec.ts/01-user-output.txt:1]: Note that we now print inserted imports. Other program level changes (e.g. `Options.gating`, outlining) should work as expected without hardcoding

File: compiler/apps/playground/__tests__/e2e/__snapshots__/page.spec.ts/parse-flow-output.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { c as _c } from "react/compiler-runtime";
+function useFoo(propVal) {
+  const $ = _c(2);
+  const t0 = (propVal.baz: number);
```
Comment (mofeiZ) [inline compiler/apps/playground/__tests__/e2e/__snapshots__/page.spec.ts/parse-flow-output.txt:4]: Flow now codegens and prints successfully. This means that Flow users can use playground without needing to edit / manually strip their annotations

File: compiler/apps/playground/__tests__/e2e/__snapshots__/page.spec.ts/parse-typescript-output.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+function Foo() {
+  const $ = _c(2);
+  let t0;
+  if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
+    t0 = foo();
+    $[0] = t0;
+  } else {
+    t0 = $[0];
+  }
+  const x = t0 as number;
```
Comment (mofeiZ) [inline compiler/apps/playground/__tests__/e2e/__snapshots__/page.spec.ts/parse-typescript-output.txt:11]: Typescript also codegens successfully 🎉 No type annotation changes needed

**Review by josephsavona — APPROVED** [review]
nice, i love that all the logs are going through the environment now, it makes it easier to log more

**Pull request merged**